common/issue-credential/src/utils.rs

```diff
@@ -0,0 +1,141 @@
+use log::*;
+use std::path::PathBuf;
+use std::process::exit;
+use std::time::{Duration, SystemTime};
+
+use nym_bandwidth_controller::acquire::state::State;
+use nym_client_core::config::disk_persistence::CommonClientPaths;
+use nym_config::DEFAULT_DATA_DIR;
+use nym_credential_storage::persistent_storage::PersistentStorage;
+use nym_validator_client::nyxd::traits::DkgQueryClient;
+use nym_validator_client::nyxd::Coin;
+use nym_validator_client::nyxd::DirectSigningNyxdClient;
+use nym_validator_client::Client;
+
+use crate::errors::Result;
+use crate::recovery_storage::RecoveryStorage;
+
+const SAFETY_BUFFER_SECS: u64 = 60; // 1 minute
+
+pub async fn issue_credential(
+    client: nym_validator_client::Client<DirectSigningNyxdClient>,
+    amount: Coin,
+    persistent_storage: &PersistentStorage,
+    recovery_storage_path: PathBuf,
+) {
```

Comment:
Since you have defined a `CredentialClientError`, perhaps return a `Result<(), CredentialClientError>` instead? .....